pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -688,7 +688,8 @@ func (c *VSphereController) applyClusterCSIDriverChange(
 		csiConfigString = strings.ReplaceAll(csiConfigString, pattern, value)
 	}
 
-	csiConfig, err := iniv1.Load([]byte(csiConfigString))
+	// We ignore inline comments when reading the INI file because passwords may contain comment symbos (# and ;)
+	csiConfig, err := iniv1.LoadSources(iniv1.LoadOptions{IgnoreInlineComment: true}, []byte(csiConfigString))
```

Comment:
for a uninitiated reader, can you add a comment that describes what this does?

Reply:
Did you see the comment above this line? Or do you mean something else?

Reply:
No that was it. I seems to have missed it. You mentioned you wanted some unit test for this stuff before we merge?

Reply:
Yup, done. PTAL